software/controller/lib/hal/adc.cpp

```diff
@@ -50,10 +50,14 @@ limitations under the License.
 //
 ////////////////////////////////////////////////////////////////////
 
+// Reference abbreviations ([RM], [PCB], etc) are defined in hal/README.md
+
 #include "adc.h"
 
 #if defined(BARE_STM32)
 
+#include <algorithm>
+
 #include "clocks.h"
 #include "dma.h"
 #include "gpio.h"
```

Comment:
So, this is no longer a dependency of ADC as reimplemented, though I am not yet convinced this is for the better?